task_DEV_4/task_DEV_4/Triangle.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+
+namespace task_DEV_4
+{
+    class Triangle : Figure
+    {
+        private const string MASSEGE_ERROR = "The triangle cannot be created because the sum of the two sides is less than the third.";
+        public double BaseSide { get; set; }
+        public double FirstSide { get; set; }
+        public double SecondSide { get; set; }
+
+        public Triangle(string name, string color, double baseSide, double firstSide, double secondSide) : base(color, name)
+        {            
+            BaseSide = baseSide;
+            FirstSide = firstSide;
+            SecondSide = secondSide;
+            CheckForExistance();
+        }
+
+        public override double CalculateArea()
+        {
+            double perimeter = CalculatePerimeter();
+            double semiPerimeter = perimeter / 2;
+            return Math.Sqrt(semiPerimeter * (semiPerimeter - BaseSide) * 
+                (semiPerimeter - FirstSide) * (semiPerimeter - SecondSide));
+        }
+
+        public override double CalculatePerimeter()
+        {
+            return BaseSide + FirstSide + SecondSide;
+        }
+        public override void CheckForExistance()
+        {
+            if (BaseSide + FirstSide <= SecondSide || SecondSide + FirstSide <= BaseSide ||
```

Comment:
What about check if side is less than 0?

Reply:
done